spec/form_models/jobseekers/job_application/professional_status_form_spec.rb

```diff
@@ -1,17 +1,169 @@
 
-  it { is_expected.to validate_inclusion_of(:qualified_teacher_status).in_array(%w[yes no on_track]) }
-  it { is_expected.to validate_inclusion_of(:statutory_induction_complete).in_array(%w[yes no]) }
+  describe "teacher reference information" do
+    subject(:form) { described_class.new(attributes) }
+
+    context "with TRI attributes explicitly provided during the initialisation" do
+      let(:attributes) { { teacher_reference_number: "1234567", has_teacher_reference_number: "yes" } }
+
+      it "sets the TRI attributes" do
+        expect(form).to have_attributes(teacher_reference_number: "1234567",
+                                        has_teacher_reference_number: "yes")
+      end
+
+      context "when a profile with different TRI attributes is also provided" do
+        let(:profile) do
+          instance_double(JobseekerProfile, teacher_reference_number: "7654321", has_teacher_reference_number: "yes")
+        end
+        let(:attributes) { super().merge(jobseeker_profile: profile) }
+
+        it "sets the explicitly provided TRI attributes over the profile attributes" do
+          expect(form).to have_attributes(teacher_reference_number: "1234567",
+                                          has_teacher_reference_number: "yes")
+        end
+      end
+    end
+
+    context "when no TRI attributes are provided" do
+      let(:attributes) { {} }
+
+      it "does not set the TRI attributes" do
+        expect(form).to have_attributes(teacher_reference_number: nil,
+                                        has_teacher_reference_number: nil)
+      end
+
+      context "with a jobseeker profile" do
+        let(:attributes) { { jobseeker_profile: profile } }
 
-  context "when qualified_teacher_status is yes" do
-    before { allow(subject).to receive(:qualified_teacher_status).and_return("yes") }
+        context "with TRI values" do
+          let(:profile) do
+            instance_double(JobseekerProfile, teacher_reference_number: "7654321", has_teacher_reference_number: "yes")
+          end
 
-    it { is_expected.to validate_numericality_of(:qualified_teacher_status_year).is_less_than_or_equal_to(Time.current.year) }
+          it "sets the teacher reference number and has_teacher_reference_number from the profile" do
+            expect(form).to have_attributes(teacher_reference_number: "7654321",
+                                            has_teacher_reference_number: "yes")
+          end
+        end
+
+        context "without TRI values" do
+          let(:profile) do
+            instance_double(JobseekerProfile, teacher_reference_number: nil, has_teacher_reference_number: nil)
+          end
+
+          it "does not set the TRI attributes" do
+            expect(form).to have_attributes(teacher_reference_number: nil,
+                                            has_teacher_reference_number: nil)
+          end
+        end
+
+        context "when explicitly stating no teacher reference number is provided" do
```

Comment:
[Minor] This made me think that we were passing a jobseeker_profile but also explicitly setting has_teacher_reference_number: "no" in the attributes. Might just be me but I would understand the test more easily something like "when the jobseeker profile has_teacher_reference_number set to no"